test/smartwallet/smartWallet.test.ts

```diff
@@ -1,943 +1,537 @@
+            const smartWallet = await smartWalletFactory.deploy();
+            const [owner, worker, utilSigner] = await hardhat.getSigners();
+        
+            return {smartWallet, owner, worker, utilSigner};
+        }
+
+        beforeEach(async function(){
+            fakeToken = await smock.fake('ERC20');
+            fakeToken.transfer.returns(true);
+        });
+
+        it('Should initialize with the correct parameters', async function () {
+            const  {smartWallet, owner, worker} = await loadFixture(prepareFixture);
+
+            await smartWallet.initialize(owner.address, fakeToken.address, worker.address, 10, 400000);
+            expect(await smartWallet.isInitialized(), 'Contract not initialized').to.be.true;            
+        });
+
+        it('Should call transfer on not sponsored deployment', async function(){
+            const {smartWallet, owner, worker} = await loadFixture(prepareFixture);
+
+            await smartWallet.initialize(owner.address, fakeToken.address, worker.address, 10, 400000);
+            expect(fakeToken.transfer).to.be.called;
+        });
+
+        it('Should not call transfer on sponsored deployment', async function(){
+            const {smartWallet, utilSigner, worker} = await loadFixture(prepareFixture);
+
+            await smartWallet.initialize(utilSigner.address, fakeToken.address, worker.address, 0, 0);
+
+            expect(fakeToken.transfer).not.to.be.called;
+        });
+
+        it('Should fail to initialize a contract when it is already initialized', async function(){
+            const {smartWallet, owner, worker} = await loadFixture(prepareFixture);
+
+            await smartWallet.initialize(owner.address, fakeToken.address, worker.address, 10, 400000);  
+
+            await expect(
+                smartWallet.initialize(owner.address, fakeToken.address, worker.address, 10, 400000),
+                'Second initialization not rejected'
+            ).to.be.revertedWith('already initialized');
+        });
+
+        it('Should create the domainSeparator', async function () {
+            const  {smartWallet, owner, worker} = await loadFixture(prepareFixture);
+
+            await smartWallet.initialize(owner.address, fakeToken.address, worker.address, 10, 400000);        
+
+            expect(await smartWallet.domainSeparator()).to.be.properHex(64);
+        });
+
+        it('Should fail when the token transfer method fails', async function () {
+            const  {smartWallet, owner, worker} = await loadFixture(prepareFixture);
+
+            fakeToken.transfer.returns(false);
+
+            await expect(
+                smartWallet.initialize(owner.address, fakeToken.address, worker.address, 10, 400000),
+                'Deployment should be reverted'
+            ).to.be.revertedWith('Unable to pay for deployment');
+        });
+    });
+
+    describe('Function verify()', function(){
+        const HARDHAT_CHAIN_ID = 31337;
```

Comment:
We should get the chain ID from the blockchain itself.

Reply:
As explained [here](https://stackoverflow.com/questions/66923541/get-chain-name-in-hardhat-not-chainid) and other posts on the Internet, there is not a simple way to obtain the chain Id from the blockchain. The mechanisms required implies external libraries and configurations. I think the work required is not worth it.